KubeArmor/enforcer/runtimeEnforcer.go

```diff
@@ -125,31 +127,44 @@ func NewRuntimeEnforcer(node tp.Node, logger *fd.Feeder) *RuntimeEnforcer {
 	re := &RuntimeEnforcer{}
 	re.Logger = logger
 
+	lsms := []string{}
+
+	lsmFile := []byte{}
+	lsmPath := "/sys/kernel/security/lsm"
+
 	if !kl.IsK8sLocal() {
 		// mount securityfs
 		if err := kl.RunCommandAndWaitWithErr("mount", []string{"-t", "securityfs", "securityfs", "/sys/kernel/security"}); err != nil {
 			if _, err := os.Stat(filepath.Clean("/sys/kernel/security")); err != nil {
-				re.Logger.Errf("Failed to read /sys/kernel/security (%s)", err.Error())
-				return nil
+				re.Logger.Warnf("Failed to read /sys/kernel/security (%s)", err.Error())
+				goto probeBPFLSM
 			}
 		}
 	}
 
-	lsm := []byte{}
-	lsmPath := "/sys/kernel/security/lsm"
-
 	if _, err := os.Stat(filepath.Clean(lsmPath)); err == nil {
-		lsm, err = os.ReadFile(lsmPath)
+		lsmFile, err = os.ReadFile(lsmPath)
 		if err != nil {
-			re.Logger.Errf("Failed to read /sys/kernel/security/lsm (%s)", err.Error())
-			return nil
+			re.Logger.Warnf("Failed to read /sys/kernel/security/lsm (%s)", err.Error())
+			goto probeBPFLSM
+		}
+	}
+
+	lsms = strings.Split(string(lsmFile), ",")
+
+probeBPFLSM:
+	if !kl.ContainsElement(lsms, "bpf") {
+		err := features.HaveProgramType(ebpf.LSM)
+		if err == nil {
+			lsms = append(lsms, "bpf")
+		} else {
+			re.Logger.Warnf("BPF LSM not supported %s", err.Error())
 		}
 	}
```

Comment:
> Remove dependency for /sys/kernel/security/lsm to check for BPF LSM. This should help things work for Minikube and KIND. Yet to be tested on KIND.
 
I didn't understand this. Here we are checking if "bpf" string is found in /sys/kernel/security/lsm and in case of minikube this file doesn't exist. How will this work in minikube?

Reply:
It's a not (`<!>kl.Contains`) check. We skip the following check if we already found bpf in the said file. We should only check for support using this method iff we didn't find the lsm file or the lsm file didn't have bpf.

Does that make sense?